microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/v20/MicroProfileOpenApi20Test.java

```diff
@@ -0,0 +1,224 @@
+                "\"/districts/{code20}\" \"response\" for GET verb has \"application/json\" but empty \"schema\" property",
+                requestBodySchema.isEmpty());
+
+        Assert.assertEquals("Specified schema for the request body is not equal to DistrictEntity",
+                "#/components/schemas/DistrictEntity", requestBodySchema.get("$ref"));
+
+        // 2.0 The @APIResponseSchema annotation has been added to provide a shorthand mechanism to specify the schema for a response body
+        Map<String, Object> patchResponses = (Map<String, Object>) patchMethod.get("responses");
+        Assert.assertFalse("\"/districts/{code20}\" \"responses\" for PATCH verb is empty", patchResponses.isEmpty());
+
+        Map<String, Object> patchHttp200Response = (Map<String, Object>) patchResponses.get("200");
+        Assert.assertFalse("\"/districts/{code20}\" \"response\" for PATCH verb and HTTP status 200 is empty",
+                patchHttp200Response.isEmpty());
+
+        Map<String, Object> patchContentAnnotation = (Map<String, Object>) patchHttp200Response.get("content");
+        Assert.assertFalse(
+                "\"/districts/{code20}\" \"response\" for PATCH verb and HTTP status 200 has empty \"content\" property",
+                patchContentAnnotation.isEmpty());
+
+        Map<String, Object> patchContentTypeJson = (Map<String, Object>) patchContentAnnotation.get("application/json");
+        Assert.assertFalse(
+                "\"/districts/{code20}\" \"response\" for PATCH verb and HTTP status 200 has \"content\" but empty \"application/json\" property",
+                patchContentTypeJson.isEmpty());
+
+        Map<String, Object> responseSchema = (Map<String, Object>) patchContentTypeJson.get("schema");
+        Assert.assertFalse(
+                "\"/districts/{code20}\" \"response\" for PATCH verb has \"application/json\" but empty \"schema\" property",
+                responseSchema.isEmpty());
+
+        Assert.assertEquals("Specified schema for the response body is not equal to DistrictEntity",
+                "#/components/schemas/DistrictEntity", responseSchema.get("$ref"));
+
+        // 2.0 The mp.openapi.schema.* property - allow the schema for a specific class to be specified.
+
+        // let's check whether there is a MockString schema in "components" property
+        Map<String, Object> components = (Map<String, Object>) yamlMap.get("components");
+        Assert.assertFalse("\"components\" property is empty", components.isEmpty());
+
+        Map<String, Object> schemas = (Map<String, Object>) components.get("schemas");
+        Assert.assertFalse("\"schemas\" property is empty", schemas.isEmpty());
+
+        Map<String, Object> mockString = (Map<String, Object>) schemas.get("MockString");
+
+        Assert.assertEquals("Schema description for String class has an unexpected value",
+                "Mock custom String class defined with config", mockString.get("description"));
```

Comment:
Maybe this can be a string constant?

Reply:
I'd like to specify this one. 

IIUC _Mock custom String class defined with config_ string could be declared as a constant in e.g. new _DistrictsResourceConstants_ class and then used in [213th line](https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/234/files#diff-54f693a93a1fc8c0db8d279a6f2a2614a217a4a6736c3747e847e0ae758b922aR213).

But as much as I know we won't be able to use this constant in [.properties file](https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/234/files#diff-0bce9f9d96ff6ea70307296cee86c6d65714484d22f804b93daad25d41336d6cR4) - is it right?

Reply:
Sure, we wouldn't. But Then if you'll need to use that constant somewhere else in the Java code you'll have to type it again.
Anyway not a deal at all, it can be done when we'll need it.